dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/event/RumEventDeserializer.kt

```diff
@@ -0,0 +1,101 @@
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain.event
+
+import com.datadog.android.core.internal.domain.Deserializer
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.rum.model.ActionEvent
+import com.datadog.android.rum.model.ErrorEvent
+import com.datadog.android.rum.model.ResourceEvent
+import com.datadog.android.rum.model.ViewEvent
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import com.google.gson.JsonSyntaxException
+
+internal class RumEventDeserializer : Deserializer<RumEvent> {
+
+    // region Deserializer
+
+    override fun deserialize(model: String): RumEvent? {
+        return try {
+            val jsonObject = JsonParser.parseString(model).asJsonObject
+            val userAttributes: MutableMap<String, Any?> = mutableMapOf()
+            val globalAttributes: MutableMap<String, Any?> = mutableMapOf()
+            val customTimings: MutableMap<String, Long> = mutableMapOf()
+            resolveAttributes(userAttributes, globalAttributes, customTimings, jsonObject)
+            RumEvent(
+                fromJson(jsonObject.getAsJsonPrimitive(EVENT_TYPE_KEY_NAME).asString, model),
+                globalAttributes,
+                userAttributes,
+                if (customTimings.isNotEmpty()) {
+                    customTimings
+                } else {
+                    null
+                }
+            )
+        } catch (e: JsonSyntaxException) {
+            sdkLogger.e("Error while trying to deserialize the serialized RumEvent: $model")
+            null
+        }
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun resolveAttributes(
+        userAttributes: MutableMap<String, Any?>,
+        globalAttributes: MutableMap<String, Any?>,
+        customTimings: MutableMap<String, Long>,
+        jsonObject: JsonObject
+    ) {
+        val customGlobalAttributesPrefix = RumEventSerializer.GLOBAL_ATTRIBUTE_PREFIX + '.'
+        val customUserAttributesPrefix = RumEventSerializer.USER_ATTRIBUTE_PREFIX + '.'
+        val customTimingsAttributesPrefix =
+            RumEventSerializer.VIEW_CUSTOM_TIMINGS_ATTRIBUTE_PREFIX + '.'
+        val customGlobalAttributesPrefixLength = customGlobalAttributesPrefix.length
+        val customUserAttributesPrefixLength = customUserAttributesPrefix.length
+        val customTimingsAttributesPrefixLength = customTimingsAttributesPrefix.length
+
+        jsonObject.keySet().forEach {
+            if (it.startsWith(customGlobalAttributesPrefix)) {
+                globalAttributes[it.substring(customGlobalAttributesPrefixLength)] =
+                    jsonObject.get(it)
+            }
+            if (it.startsWith(customUserAttributesPrefix)) {
+                userAttributes[it.substring(customUserAttributesPrefixLength)] =
+                    jsonObject.get(it)
+            }
```

Comment:
Looking at those constants:

https://github.com/DataDog/dd-sdk-android/blob/4ba301f58e9b2605062bcbfc4991ba142a4ac22f/dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/event/RumEventSerializer.kt#L103-L104

the order of those two checks seems to be important, as `customGlobalAttributesPrefix` is a superset of `customUserAttributesPrefix` (the `context.*` contains all of `context.usr.*`). I think we need to reverse them, no?

Reply:
Hmm....you are right, otherwise we might assign different user attributes to global. I need to see how can I add a test case for that.